### ISSUE TITLE
Convert extension Welcome pages to Getting Started walkthroughs

### ISSUE THREAD

**Issue by digitarald** [microsoft/vscode#118402]
To further refine #118055, we need to convert some existing _welcome_ pages _(aka: anything that extensions pop up on a fresh install)_ to use the proposed getting started API. We expect that this migration can a) make start pages more contextual and timely, and b) provide organic discovery for extension content.

Each extension can roughly follow these steps:

1. **Walkthrough content draft.** Easier to collaborate and iterate on the overall journey and detailed copy.
2. **Ship `walkthrough` declarations to release.** Until VS Code enables extension walkthroughs, these will not be visible to users but will be ready for usertesting & experiment steps.
3. Test E2E scenarios
   - New and returning user in VS Code
   - Remote & Codespaces
4. In parallel:
    1. **Usertesting each walkthrough**. Ask participants to enable `workbench.welcomePage.experimental.extensionContributions` on Insiders before guiding them through the usual language/extension setup.
    2. **A/B experiment each walkthrough**. Use the flags provided by #122966 to enable walkthroughs to run independent experiments for each extension walkthrough. If an extension has custom welcome page, the same flags need to be used to disable those.
       - Independent of experiments on each walkthrough VS Code can run experiments on when to show walkthroughs and other tweaks to the experience.

### Progress

To start with, we are looking into extensions that have been coming up during usertesting:

- [x] Python ([Start page](https://github.com/microsoft/vscode-python/tree/main/src/startPage-ui/startPage))
   - [x] First conversations: Ongoing work on an updated Welcome page that.
   - [x] Shared UX template
   - [x] Prototype the current Welcome content in Getting Started
   - [x] Land walkthrough in Insiders: Blocked by providing a when clause for python detection.
   - [x] Shipped
   - [x] Test E2E scenario (vscode new and returning user, remote, codespaces)
   - [x] Usertesting
   - [x] Experiment
   - [x] Done
- [x] Java ([Getting started](https://github.com/microsoft/vscode-java-pack/tree/master/src/getting-started), [Welcome](https://github.com/microsoft/vscode-java-pack/tree/master/src/welcome), [Extension guide](https://github.com/microsoft/vscode-java-pack/tree/master/src/ext-guide))
   - [x] First prototype completed
   - [x] Shipped
   - [x] Usertesting
   - [x] Experiment
   - [x] Done
- [x] C++ (no welcome page)
   - [x] Kick-off meeting
   - [x] Draft content proposal
   - [x] Shipped
   - [x] Test E2E scenario (vscode new and returning user, remote, codespaces)
   - [x] Usertesting
   - [x] Experiment
   - [x] Done
- [ ] WSL ([Welcome page](https://github.com/microsoft/vscode-internalbacklog/issues/1320#issuecomment-713706235) on hold)
   - [ ] Draft content
   - [ ] Shipped
   - [ ] Usertesting
   - [ ] Experiment
   - [ ] Done
- [ ] [TypeScript](https://github.com/microsoft/vscode/issues/121345)
   - [ ] Prototype content
   - [ ] Shipped
   - [ ] Test E2E scenario (vscode new and returning user, remote, codespaces)
   - [ ] Usertesting
   - [ ] Experiment
   - [ ] Done
- [ ] 💪 Live Share
   - [ ] Prototype content
   - [ ] Shipped
   - [ ] Usertesting
   - [ ] Experiment
   - [ ] Done
- [x] 💪 Docker ([Start page](https://github.com/microsoft/vscode-docker/blob/main/resources/startPage.html.template) running as experiment right now)
   - [x] Draft content
   - [x] Shipped
   - [x] Usertesting
   - [x] Experiment
   - [x] Done
- [x] 🛑 ~Azure~
- [x] 🛑 ~.NET (_ongoing work_)~
   - [x] Kick-off meeting

### Open Blockers

- https://github.com/microsoft/vscode/issues/119725
- https://github.com/microsoft/vscode/issues/119709

### Learnings

- Opening files, build-in pages and projects without yanking users out of getting started
  - Commands and flows that open new tabs need to be opened side-by-side with getting started.
  - New projects currently open a new window. Could either open in current window (if empty) or with getting started restored.
- Most extension Welcome pages offer focus on the multiple ways to get started: _Open folder_, _Open file_, _New file_, _New file type x_ (from template), _Create project_ (from wizard), etc
   - How do we expose in _Getting Started_, which follows a serial content pattern
   - Contribute those entry points in Welcome page and Explorer

**Comment by digitarald** [comment]
Demo for the Java extension ([pull request](https://github.com/microsoft/vscode-java-pack/pull/588)) as example:

https://user-images.githubusercontent.com/8599/113540150-55ac8480-9594-11eb-8ba9-7f3f10b2283a.mp4

**Comment by sean-mcmanus** [comment]
@digitarald With our C/C++ extension we're seeing "when": "isWindows" giving results for the local machine when we need the remote machines. Should we file a VS Code bug or is there a different "when" clause we should be using.

```
    "walkthroughs": [
      {
...
        "steps": [
          {
            "when": "isMac",
...
```

**Comment by luabud** [comment]
cc @kimadeline

**Comment by digitarald** [comment]
As most extensions shipped walkthroughs, breaking this down into smaller follow up items.
